velox/functions/sparksql/aggregates/MinMaxAggregate.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/aggregates/MinMaxAggregateBase.h"
+
+namespace facebook::velox::functions::aggregate::sparksql {
+
+namespace {
+
+// Spark requires timestamp data precision to be in microseconds. Add this class
+// to align with Spark's behavior.
```

Comment:
nit: comment suggestion:

Extracts timestamp value as microsecond precision in max aggregate.